src/custom/utils/price.ts

```diff
@@ -0,0 +1,213 @@
+import { BigNumber } from '@ethersproject/bignumber'
+import BigNumberJs from 'bignumber.js'
+
+import { getFeeQuote, getPriceQuote as getPriceQuoteGp, OrderMetaData } from 'utils/operator'
+import GpQuoteError, { GpQuoteErrorCodes } from 'utils/operator/errors/QuoteError'
+import { getCanonicalMarket, isPromiseFulfilled, withTimeout } from 'utils/misc'
+import { formatAtoms } from 'utils/format'
+import { PRICE_API_TIMEOUT_MS } from 'constants/index'
+import { getPriceQuote as getPriceQuoteParaswap, toPriceInformation } from 'utils/paraswap'
+
+import { OptimalRatesWithPartnerFees as OptimalRatesWithPartnerFeesParaswap } from 'paraswap'
+import { OrderKind } from '@gnosis.pm/gp-v2-contracts'
+import { ChainId } from 'state/lists/actions'
+
+const FEE_EXCEEDS_FROM_ERROR = new GpQuoteError({
+  errorType: GpQuoteErrorCodes.FeeExceedsFrom,
+  description: GpQuoteError.quoteErrorDetails.FeeExceedsFrom,
+})
+
+export interface QuoteParams {
+  quoteParams: FeeQuoteParams
+  fetchFee: boolean
+  previousFee?: FeeInformation
+  isPriceRefresh: boolean
+}
+
+export interface FeeInformation {
+  expirationDate: string
+  amount: string
+}
+
+export interface PriceInformation {
+  token: string
+  amount: string | null
+}
+
+export class PriceQuoteError extends Error {
+  params: PriceQuoteParams
+  results: PromiseSettledResult<any>[]
+
+  constructor(message: string, params: PriceQuoteParams, results: PromiseSettledResult<any>[]) {
+    super(message)
+    this.params = params
+    this.results = results
+  }
+}
+
+export type FeeQuoteParams = Pick<OrderMetaData, 'sellToken' | 'buyToken' | 'kind'> & {
+  amount: string
+  fromDecimals: number
+  toDecimals: number
+  chainId: ChainId
+}
+
+export type PriceQuoteParams = Omit<FeeQuoteParams, 'sellToken' | 'buyToken'> & {
+  baseToken: string
+  quoteToken: string
+  fromDecimals: number
+  toDecimals: number
```

Comment:
`fromDecimals` and `toDecimals` are redundant and should be removed, since they are already part of `FeeQuoteParams`